generators/client/templates/src/main/webapp/app/components/login/_login.controller.js

```diff
@@ -48,9 +48,8 @@
 
                 // If we're redirected to login, our
                 // previousState is already set in the authExpiredInterceptor. When login succesful go to stored state
-                if ($rootScope.redirected && $rootScope.previousStateName) {
-                    $state.go($rootScope.previousStateName, $rootScope.previousStateParams);
-                    $rootScope.redirected = false;
```

Comment:
Thanks @deepu105. I think the previousStateName being set acts as the redirected flag itself. I've updated the PR to also clear the previousState before redirecting to it. The PR should be passing now but I the build is broken from https://github.com/jhipster/generator-jhipster/commit/72886856a850c61ae4ef81a7c8aaf29c93433768. I'll close and reopen this once the build is fixed.